### PR TITLE
fix(handlers/webhook): return Message object instead of raw payload

### LINKED ISSUE #592: Edit Webhook Message Returns A Message Object On Success
Reference: https://github.com/discord/discord-api-docs/pull/2461

(Add code `50027`)

# PENDING

### PR DESCRIPTION
**Pull request by Skillz4Killz**
Closes #592

### CODE REVIEW

**Review by ayntee — DISMISSED** [review]
LGTM

File: src/api/handlers/webhook.ts [hunk 1]
```diff
@@ -552,5 +553,10 @@ export async function editSlashResponse(
     options,
   );
 
-  return result;
+  if (options.messageID) return result;
+
+  const message = await structures.createMessage(
+    result as MessageCreateOptions,
+  );
+  return message;
```
Comment (itohatweb) [inline src/api/handlers/webhook.ts:561]: Are you sure that `Delete Original Interaction Response` returns the message?

Comment (Skillz4Killz) [reply src/api/handlers/webhook.ts:561]: Github is confusing u this line if part of `editSlashResponse`. For quick confirmation check th eline numbers

Comment (itohatweb) [reply src/api/handlers/webhook.ts:561]: Ah lol you're right

File: src/api/handlers/webhook.ts [hunk 2]
```diff
@@ -552,5 +553,10 @@ export async function editSlashResponse(
     options,
   );
 
-  return result;
+  if (options.messageID) return result;
```
Comment (itohatweb) [inline src/api/handlers/webhook.ts:556]: `Edit Webhook Message` does return a message object too, so why not using `structures.createMessage` here too?

Comment (Skillz4Killz) [reply src/api/handlers/webhook.ts]: Good catch that should have been if (!options.messageID

File: src/api/handlers/webhook.ts [hunk 3]
```diff
@@ -552,5 +553,11 @@ export async function editSlashResponse(
     options,
   );
 
-  return result;
+  // If the original message was edited, his will not return a message
```
Comment (itohatweb) [inline src/api/handlers/webhook.ts:556]:
```suggestion
  // If the original message was edited, this will not return a message
```

**Review by itohatweb — CHANGES_REQUESTED** [review]

**Review by ayntee — APPROVED** [review]
LGTM

**Review by itohatweb — APPROVED** [review]

**Pull request merged**